### PR TITLE
[REVIEW] Add Java and JNI bindings for string contains [skip ci]

### PR DESCRIPTION
**Pull request by kuhushukla**
Description: Add java and JNI bindings for string `contains` in find operations (`find.cu`) - similar to `startsWith` and `endsWith` bindings.

### CODE REVIEW

**Comment by kuhushukla** [comment]
@revans2 Please take a look. Thank you!

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -1744,6 +1744,23 @@ public ColumnVector endsWith(Scalar pattern) {
...
   }
 
+  /**
+   * Checks if each string in a column contains a specified comparison string, resulting in a
+   * parallel column of the boolean results.
+   * @param pattern scalar containing the string being searched for.
+   * @return A new java column vector containing the boolean results.
+   */
+
+  public ColumnVector stringContains(Scalar pattern) {
```
Comment (razajafri) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: nit: I might be missing something but looking at the cpp header, its just a regular string, calling it a pattern might throw people off?

Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: This keeps it in line with startswith and endswith implementations. Let me know if that works or I'll go ahead and change all three although that would expand the footprint of the patch.

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: I do think it should be changed but I will leave that up to you.

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: LGTM otherwise

Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Thanks Reza for the feedback. I would keep it this way and maybe do a refactor PR separately to keep commit annotations clean.

**Comment by kuhushukla** [comment]
> Looks good to me, very similar to the starts/ends with functions. Small logging message typo that is being fixed.

Addressed. Thanks for catching that

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -1744,6 +1744,23 @@ public ColumnVector endsWith(Scalar pattern) {
...
+    assert type == DType.STRING : "column type must be a String";
+    assert pattern != null : "pattern scalar may not be null";
+    assert pattern.getType() == DType.STRING : "pattern scalar must be a string scalar";
+    assert pattern.isValid() == true : "pattern string scalar may not contain a null value";
+    assert pattern.getJavaString().isEmpty() == false : "pattern string scalar may not be empty";
```
Comment (mythrocks) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Nit:
```java
assert pattern.isValid() : "pattern string scalar may not contain a null value";
assert !pattern.getJavaString().isEmpty() : "pattern string scalar may not be empty";
```

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 1 above)
Comment (mythrocks) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Hmm. @razajafri might have a point here. I'd think `pattern` has regex-like connotations.

Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: I agree, I did not want to change `startsWith` and `endsWith` in a PR which adds `contains`. Let me just change it for the method I am working on and do the others as a follow up. Thanks @razajafri / @mythrocks

**Comment by kuhushukla** [comment]
@mythrocks / @razajafri / @rwlee , please take another look as I think it is ready to go.

**Comment by kuhushukla** [comment]
@kkraus14 , may I please request you to admin merge this PR? Thanks a lot!
FYI ci was skipped.

**Comment by kuhushukla** [comment]
@kkraus14 , Could you merge this PR? Appreciate it! CI was skipped on this.

**Comment by kuhushukla** [comment]
@kkraus14 , Could we admin merge this PR please? Thanks a lot!

**Pull request merged**